CMakeLists.txt

```diff
@@ -8,17 +18,19 @@ IF (NOT TARGET ns_realclean)
   MESSAGE(FATAL_ERROR "ns_realclean target does not exist.")
 ENDIF ()
 
-SET (REBAR_OPTS -C "${CMAKE_CURRENT_SOURCE_DIR}/rebar2.config")
-
 REBAR (TARGET chronicle REBAR_OPTS ${REBAR_OPTS} CLEAN_HOOK ns_realclean)
 REBAR (TARGET chronicle_dump REBAR_OPTS ${REBAR_OPTS}
-  COMMAND escriptize NOCLEAN DEPENDS chronicle)
+       COMMAND escriptize NOCLEAN DEPENDS chronicle)
 
 SET(INSTALL_DIR "${INST_LIBDIR}/chronicle")
 
-# This assumes that rebar2 is used.
-INSTALL (DIRECTORY ebin DESTINATION "${INSTALL_DIR}")
-INSTALL (PROGRAMS chronicle_dump DESTINATION bin)
+IF (${REBAR_VERSION} STREQUAL "rebar2")
+  INSTALL (DIRECTORY ebin DESTINATION "${INSTALL_DIR}")
+  INSTALL (PROGRAMS chronicle_dump DESTINATION bin)
+ELSE()
+  INSTALL (DIRECTORY _build/default/lib/chronicle/ebin DESTINATION "${INSTALL_DIR}")
+  INSTALL (PROGRAMS _build/default/bin/chronicle_dump DESTINATION bin)
+ENDIF()
 
 # priv/ only exists on Linux currently, so make this OPTIONAL
 INSTALL (DIRECTORY priv OPTIONAL DESTINATION "${INSTALL_DIR}")
```

Comment:
Probably need to handle priv/ here. Need to compile on linux to test.

Reply:
Seems like this is fine actually. Rebar3 is compiling priv/ to the same place and just symlinking it into the rebar3 build directory. Given that we are remapping the installation directory in our install step I think that this should just be fine.